main.cpp

```diff
@@ -201,6 +201,51 @@ Thing 1) Coffee maker
     3) beep when pouring complete
  */
 
+struct CoffeeMaker
+{
+    //maximum water temperature in F
+    float maxWaterTemp = 75;
+    //cup size in mls
+    int cupSize = 200;
+    //indicate whether cup full
+    bool cupFull = false;
+    //type of coffee (espresso/cappuccino/etc)
+    std::string typeOfCoffee = "espresso";
+    //amount of water remaining in reservoir in mls
+    float waterRemaining = 500;
+
+    struct Cup
+    {
+        //cup volume in mls
+        float cupVolume = 250;
+        //cup material
+        std::string material = "glass";
+        //cup height in cm
+        float cupHeight = 12;
+        //cup colour
+        std::string cupColour = "black";
+        //indicate whether cup clean
+        bool cupClean = true;
+
+        //fill cup
+        void setCupVolume(float cupVolume);
+        //get fill status
+        bool getCleanStatus(bool cupClean); //indcates whether cup clean or not
+        //set fill status
+        void setCupColour(std::string cupColour);
+    };
+
+    //heat water
+    void heatWater();
+    //deliver correct quantity of heated water
+    void fillCup(int cupSize);
+    //beep when pouring complete
+    void beepWhenCupFull(bool cupFull);
```

Comment:
We already know it is a cup. We don't need the word `cup` in the variable names.
- [ ] revise